x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java

```diff
@@ -205,6 +205,13 @@ public void testCreateAndRestoreSearchableSnapshot() throws Exception {
         } else {
             expectedDataTiersPreference = getDataTiersPreference(MountSearchableSnapshotRequest.Storage.FULL_COPY);
         }
+        final String indexCheckOnStartup;
+        if (randomBoolean()) {
+            indexCheckOnStartup = randomFrom("false", "true", "checksum");
+            indexSettingsBuilder.put(IndexSettings.INDEX_CHECK_ON_STARTUP.getKey(), indexCheckOnStartup);
+        } else {
+            indexCheckOnStartup = "false";
```

Comment:
We set this explicitly to `false` above (line 170); I think for this to be meaningful we should not set it in the mount request and make sure it's set randomly in the original index.

Reply:
🤦🏻 

I pushed 2b8edfe5e660e1aeb49f0549464e22468036e1b1 to set a default value in the original index settings and check it is mounted with the `false` default value.

I also added a test to verify that it can be overridden in the mount request.

I also cleaned up other tests to not explicitly set the `false` value.